pkg/gui/context/sub_commits_context.go

```diff
@@ -37,15 +37,17 @@ func NewSubCommitsContext(
 	}
 
 	getDisplayStrings := func(startIdx int, length int) [][]string {
-		currentBranchName := ""
 		selectedCommitSha := ""
 		if c.CurrentContext().GetKey() == SUB_COMMITS_CONTEXT_KEY {
-			currentBranchName = viewModel.GetRef().RefName()
 			selectedCommit := viewModel.GetSelected()
 			if selectedCommit != nil {
 				selectedCommitSha = selectedCommit.Sha
 			}
 		}
+		currentBranchName := ""
+		if viewModel.GetRef() != nil {
+			currentBranchName = viewModel.GetRef().RefName()
+		}
```

Comment:
I assume the nil check is for when you've just init'd a repo and you press enter on the master branch, which contains no commits

Reply:
No, that's not it; in that case we have a non-nil ref.

It seems to be something specific to running integration tests. I still don't know how to debug integration tests, but if I put a panic here if the ref is nil, then the test `set_upstream.go` reliably crashes. It doesn't even show subcommits. However, if I manually open `test/results/branch/set_upstream/actual/repo` in lazygit and do what the test does, it doesn't crash. I'm kind of at a loss here, any more ideas?

Reply:
Ah so what's special about that test is that it changes the screen mode i.e. it maximises the current window. This calls `rerenderViewsWithScreenModeDependentContent` in `pkg/gui/controllers/screen_mode_actions.go` which renders the subcommits context which ends up calling getDisplayStrings here where there's no ref defined.

I'm thinking we should exit early from getDisplayStrings if there's no ref defined. We could add conditional logic higher up to prevent it from being rendered but I think it's fine having that logic in the context file itself

Reply:
I see. I guess I find it strange to ask a view to rerender if it has nothing to display. It looks like the gocui view is not visible in this case, so might this be a better condition for skipping it in `rerenderViewsWithScreenModeDependentContent`?

Reply:
We'd need to be careful: if we're in the files view and we maximise it, then every other view is considered hidden (whether it's actually got `Visible: false` I'm not sure but I suspect so), and so if we then focus the commits window, we'd want to make sure it's got the content we want to see.

Reply:
OK, I see. I'll go with your suggestion to early return in `getDisplayStrings` when we don't have a ref; see 9a2e9b9daf5a.